pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1684,32 +1703,43 @@ init_context(event_context_t *context, struct task_struct *task, u32 options)
     return 0;
 }
 
-static __always_inline task_info_t *init_task_info(u32 key, bool *initialized)
+static __always_inline task_info_t *init_task_info(u32 tid, u32 pid, bool *initialized)
```

Comment:
Perhaps a signature like `init_task_info(u64 pid_tgid, bool *initialized)`, then in its scope making use of helpers to access the tgid and pid. I'm mentioning that because I noticed that it's being done manually by callers: `task_info_t *task_info = init_task_info(pid_tgid, pid_tgid >> 32, NULL);`

Reply:
yes, that's a good idea!

Reply:
Actually there is one caller that doesn't have pid_tgid ready, and it needs to encode it.
So I'm not sure it will be prettier